### ISSUE TITLE
Project reference paths are null in d15prerel (makes E2E assertions fail)

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/Home#3783]
This seems to be a regression in DTE/Visual Studio, but I am tracking this here until I know more.

Working VS versions:
d14u3, d15p5, d15prerel/25819.0

Broken VS versions:
d15prerel/25825.4
## Steps
1. Open Visual Studio (I used d15prerel/25825.4)
2. File new project a legacy C# console app.
3. Open PMC
4. `Install-Package Newtonsoft.Json -Version 8.0.3`
5. `$project = Get-Project`
6. `$reference = $project.Object.References.Item("Newtonsoft.Json")`
7. `$reference.Path`
## Expected

The full path for the Newtonsoft.Json DLL should be printed in PMC.
## Actual

Nothing is printed
## Failing tests

```
BindingRedirectsMixNonStrongNameAndStrongNameAssemblies
DiamondDependencies
InstallPackageAddImportStatement
InstallPackageRespectAssemblyReferenceFilterOnDependencyPackages
InstallPackageRespectAssemblyReferenceFilterOnSecondProject
InstallPackageTargetingNetClientAndNet
InstallPackageWithExeReference
InstallPackageWithGacReferencesIntoMultipleProjectTypes
InstallPackageWithNestedReferences
InstallPackageWithReferences
InstallPackageWithResourceAssemblies
PackageInstallAcceptsRelativePathSource
PackageInstallAcceptsRelativePathSource2
PackageInstallAcceptsSourceName
PackageWithNoVersionInFolderName
RemovePackageRemovesPackageFromSolutionIfNotInUse
RemovingPackageFromProjectDoesNotRemoveIfInUse
SimpleBindingRedirects
SimpleBindingRedirectsClassLibraryReference
SimpleBindingRedirectsClassLibraryUpdatePackage
SinglePackageInstallIntoSingleProject
UninstallingPackageWithConfigTransformWhenConfigReadOnly
UninstallPackageUseTargetFxPersistedInPackagesConfigToRemoveAssemblies
UninstallPackageWhatIf
UpdatingPackageInProjectDoesntRemoveFromSolutionIfInUse
```

**Comment by rrelyea** [comment]
Fix in PR by Tom

**Comment by mishra14** [comment]
From Tom -
Yes. The fix for that is in d15prerel and wpt.
D15prerel is still chewing on 25827.02, which predates my change. Anything after that will have the fix.

Waiting for the build to be available.

**Comment by joelverhagen** [comment]
Ankit and I validated this fix with 25830.00. DTE Project Reference objects look good. Those E2E tests are now getting past this specific failure.
